test/lib/render-helpers.js

```diff
@@ -99,6 +99,27 @@ export function renderHookWithProvider(hook, state, pathname = '/', Container) {
   };
 }
 
+/**
+ * Renders a hook with a provider and optional container.
+ *
+ * @template {(...args: any) => any} Hook
+ * @template {Parameters<Hook>} HookParams
+ * @template {ReturnType<Hook>} HookReturn
+ * @template {import('@testing-library/react-hooks').RenderHookResult<HookParams, HookReturn>} RenderHookResult
+ * @template {import('history').History} History
+ * @param {Hook} hook - The hook to be rendered.
+ * @param [state] - The initial state for the store.
+ * @param [pathname] - The initial pathname for the history.
+ * @param [Container] - An optional container component.
+ * @returns {RenderHookResult & { history: History }} The result of the rendered hook and the history object.
```

Comment:
JSDOC is actually pretty powerful. I did not know we can utilise generics.
This give us a nicer return type which gives us correct intellisense when creating tests.
```
const hook = renderWithProviderTyped(() => useMyFoo())
hook.result.callBar // <-- this will now autocomplete and give us info on return/params required
```

We can migrate this over to TS in a diff PR (along with the other render helper functions in this file).

Reply:
As for why not adding JSDOC to the original function - I tested this, but many other tests that used this function failed TSC (due to not using the correct `result.current` type or passing incorrect arguments)

Reply:
I think we can get a follow-up PR to add the JSDOC types to the original function, and fix the type issues on other files.

It might require coordinating with other teams to make sure that the type fixes are expected.